lambdas/chromium/screenshot.js

```diff
@@ -0,0 +1,131 @@
+const puppeteer = require('puppeteer-core')
+const chromium = require('chrome-aws-lambda')
+const { parse } = require('url')
+const debug = require('debug')('screenshot')
+
+const {
+  URL_WHITELIST,
+  PUPPETEER_WS_ENDPOINT
+} = process.env
+
+const DEFAULT_WIDTH = 1200
+const DEFAULT_HEIGHT = 1
+const DEFAULT_SCALE_FACTOR = 1
+const FORMATS = ['png', 'jpeg']
+
+if (!URL_WHITELIST) {
+  console.warn('missing env URL_WHITELIST, the /render endpoint will not work')
+}
+const whitelistedUrls = URL_WHITELIST && URL_WHITELIST.split(',')
+
+const getBrowser = async () => {
+  if (chromium.headless) {
+    debug('rendering with local headless chrome')
+    return puppeteer.launch({
+      args: chromium.args,
+      executablePath: await chromium.executablePath,
+      headless: chromium.headless
+    })
+  } else {
+    if (!PUPPETEER_WS_ENDPOINT) {
+      console.warn('missing env PUPPETEER_WS_ENDPOINT, cannot render')
+      return
+    }
+    debug(`rendering with chromium @ PUPPETEER_WS_ENDPOINT`)
+    return puppeteer.connect({
+      browserWSEndpoint: PUPPETEER_WS_ENDPOINT
+    })
+  }
+}
+
+const getPosInt = (number) =>
+  Math.ceil(Math.abs(number))
+
+// returns buffer
+module.exports = async (req, res) => {
+  const {
+    query: {
+      url,
+      width,
+      height,
+      zoomFactor,
+      fullPage = '1',
+      format: _format,
+      quality,
+      cookie,
+      basicAuthUser,
+      basicAuthPass
+    }
+  } = parse(req.url, true)
+
+  const format = (_format && FORMATS.indexOf(_format) !== -1)
+    ? _format
+    : 'png'
+
+  debug({ url, width, height, zoomFactor, fullPage, format, quality, cookie, basicAuthUser, basicAuthPass })
```

Comment:
```suggestion
  debug({ url, width, height, zoomFactor, fullPage, format, quality, cookie, basicAuthUser, basicAuthPass: !!basicAuthPass })
```

Would not print credentials, or at least not password.